code/modules/mob/dead/observer/say.dm

```diff
@@ -12,15 +12,18 @@
 /mob/dead/observer/Hear(message, atom/movable/speaker, message_language, raw_message, radio_freq, list/spans, message_mode)
 	. = ..()
 	var/atom/movable/to_follow = speaker
+	var/list/view_size = getviewsize(client.view)
 	if(radio_freq)
 		var/atom/movable/virtualspeaker/V = speaker
 		if(isAI(V.source))
 			var/mob/living/silicon/ai/S = V.source
 			to_follow = S.eyeobj
 		else
 			to_follow = V.source
-	else if(client && (to_follow in range(client.view)))
+
+	else if(client && get_dist(src, to_follow) < CEILING(view_size[1] / 2, 1))
```

Comment:
Why not use `in_view_range(src, to_follow)`?

Reply:
spoke offline, in_view_range wasn't doing what it should have